tests/Bugsnag.Tests/ClientTests.cs

```diff
@@ -41,5 +41,35 @@ public class ClientTests
 
       Assert.Single(requests);
     }
+
+    [Fact]
+    public async void TestNonThrownException()
+    {
+      var server = new TestServer(1);
+
+      server.Start();
+
+      var metadata = new Dictionary<string, object>() { { "password", "secret" } };
+      var subMetadata = new Dictionary<string, object>() { { "circular", metadata }, { "password", "not again!" }, { "uri", new Uri("http://google.com?password=1111&wow=cool") } };
+      metadata["test"] = subMetadata;
+      var filters = new string[] { "password" };
+
+      var client = new Client(new Configuration("123456") { Endpoint = server.Endpoint, MetadataFilters = filters });
+
+      client.BeforeNotify(r => {
+        foreach (var @event in r.Events)
+        {
+          @event.Metadata["bugsnag"] = metadata;
+        }
+      });
+
+      client.Notify(new ArgumentNullException());
+
+      var requests = await server.Requests();
+
+      var request = requests.Single();
+
+      Assert.DoesNotContain("Bugsnag.Client.Notify", request);
```

Comment:
This should assert that an expected frame is present as well.

Reply:
I looked into this but it is difficult as the method is being called inside an async xunit test so alot of the method names are auto generated continuations

Reply:
How about adding a method to this class which calls `Notify`, and having the test call that method and verify that the intermediate method is present in the request?

Reply:
I'll give it a go

Reply:
great idea, it worked!